Add custom courses page [courses]

stacyk opened the pull request:
![](https://picsum.photos/600/400)

## Description
WIP - Adds custom courses page, the most basic version similar to this mockup: 
[Figma file](https://www.figma.com/design/gJ1cVEGOMRIVgLzB3TMTbQ/Learn-List-Page?node-id=4-91&t=PiTywO01UYbF0V9m-0) 

Eventually, I think we want to go with this [medium term solution](https://www.figma.com/design/gJ1cVEGOMRIVgLzB3TMTbQ/Learn-List-Page?node-id=334-690&t=NkXFmKeKCov4Ll8D-0)

## Related Issue(s)
https://github.com/orgs/oddbird/projects/13/views/5?pane=issue&itemId=101662446&issue=oddbird%7Coddleventy%7C841

## Steps to test/reproduce

stacyk commented on content/learn/index.njk:
```diff
@@ -0,0 +1,34 @@
+
+{% else %}
+<p>No courses found.</p>
+{% endif %}
```
@jgerigmeyer Thanks! This works. I noticed that these courses are in a learn folder (so the url is /learn instead of /courses) ... should we rename that to courses? I know we called it learn originally but I think this is changing with the new nav updates.

stacyk replied:
@SondraE do you have a url preference? Basically, should the url stay learn/whatever or courses/whatever 

Right now the 2 items url paths are:
oddbird.net/learn/courses/anchor-positioning/ 
oddbird.net/learn/courses/design-with-code/

Are we going to put other content in learn outside of courses? I think this was done originally when we were going to have talks and workshops also under the "learn" menu item.

stacyk commented:
> @stacyk
> 
> > Are we going to put other content in learn outside of courses?
> 
> At this point, we aren't planning to combine content into a "learn" category, so we don't need that part of the url moving forward.

@jgerigmeyer Do you think we would want to edit any of the URLs in the email course in Mailchimp? I know the redirect will work, but is it preferable/worth the time to have that new link from the email?

stacyk commented on content/courses/index.njk:
```diff
@@ -0,0 +1,16 @@
+---
+title: Courses
+index: Course
+list_label: All Courses
+sub: Frontend web development courses for curious developers
+summary: |
+  Video and email courses for frontend web developers on modern CSS including
+  anchor positioning, grid & subgrid, container queries and more!
+---
```
@jgerigmeyer when I moved the contents of this folder up a level, the email course no longer is displayed on this `courses` page. Should it be nested in an email folder or something like the video course?

jgerigmeyer replied:
@stacyk Try again? I think I fixed it.

stacyk replied:
Thank you! That works.

jgerigmeyer commented:
> @jgerigmeyer Do you think we would want to edit any of the URLs in the email course in Mailchimp? I know the redirect will work, but is it preferable/worth the time to have that new link from the email?

Hm, that's a good question. I'm not sure how time-consuming it is to go through and update those -- @jamesnw what do you think?

stacyk commented:
> Looks good. A few tangential notes:
> 
> * add the Sara Souidian quote about James?
> * should we put my name in the Charis quote?
> * looks like we're using an email icon for all courses?

1. Replaced one testimonial with the Learn with Jason one 
2. I updated the quote to have Miriam's name
3. I added a new issue to address the icons after this PR: https://github.com/oddbird/oddleventy/issues/963

jamesnw commented:
> > @jgerigmeyer Do you think we would want to edit any of the URLs in the email course in Mailchimp? I know the redirect will work, but is it preferable/worth the time to have that new link from the email?
> 
> Hm, that's a good question. I'm not sure how time-consuming it is to go through and update those -- @jamesnw what do you think?

It would probably take me 15 minutes or so... I can do that once this is merged.

jamesnw reviewed (approved):
🚀
